### PR TITLE
New Firebase auth features

### LINKED ISSUE #58: Investigate adding new authentication features to the FirebaseAuth script
Currently we are missing small things like the ability to edit emails or passwords. Firebase has everything listed in the REST API we can do.

We should look into adding

* Change user email
* Change user password
* Reset a lost password
* Delete user account

https://firebase.google.com/docs/reference/rest/auth#top_of_page

### PR DESCRIPTION
**Pull request by BearDooks**
This change focuses on adding some new Firebase Auth features that were missing. New features added include

* Sending a password reset email
* Deleting a User Account (As in user deletes their own account)
* Change email or password (Logged in user can change these)
* Send account verification email
This closes #58

### CODE REVIEW

File: addons/godot-firebase/FirebaseAuth.gd [hunk 1]
```diff
@@ -9,6 +9,9 @@ var signup_request_url = "https://identitytoolkit.googleapis.com/v1/accounts:sig
 var signin_request_url = "https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key=%s" 
 var userdata_request_url = "https://identitytoolkit.googleapis.com/v1/accounts:lookup?key=%s" 
 var refresh_request_url = "https://securetoken.googleapis.com/v1/token?key=%s"
+var oobcode_request_url = "https://identitytoolkit.googleapis.com/v1/accounts:sendOobCode?key=%s"
+var delete_account_request_url = "https://identitytoolkit.googleapis.com/v1/accounts:delete?key=%s"
+var update_account_request_url = "https://identitytoolkit.googleapis.com/v1/accounts:update?key=%s"
```
Comment (lukehollenback) [inline addons/godot-firebase/FirebaseAuth.gd:14]: Can all of these be `const` instead of `var`?

**Review by lukehollenback — APPROVED** [review]
One thing to consider is, instead of having the following duplicated quite a bit throughout the class...

```gdscript
if (is_busy == false):
  # Fire off some request...
else:
  printerr("Firebase Auth is currently busy and cannot process this request")
```

...Consider unifying this code into a single method. For example...

```gdscript
func _is_ready() -> bool:
  if is_busy:
    printerr("Firebase Auth is currently busy and cannot process this request")
    return false
  
  return true
```

...Then all of this code can be simplified to something like...

```gdscript
if _is_ready():
  # Fire off some request...
```

**Review by WolfgangSenff — CHANGES_REQUESTED** [review]
Have to make sure you set is_busy to true before you make every request.

**Comment by lukehollenback** [comment]
> Have to make sure you set is_busy to true before you make every request.

Note that you could bake this into the `_is_ready()` method that I proposed in my review so you don't have duplicate code.

**Comment by WolfgangSenff** [comment]
> > Have to make sure you set is_busy to true before you make every request.
> 
> Note that you could bake this into the `_is_ready()` method that I proposed in my review so you don't have duplicate code.

I'd be okay with that, though I'd prefer the shorter: return not is_busy for the end. Not a big deal either way to me.

**Review by WolfgangSenff — APPROVED** [review]

**Pull request merged**